YAXLib/StringExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+#nullable enable
+using System;
+using System.Buffers;
+using System.Text;
+
+namespace YAXLib
+{
+    internal static class StringExtensions
+    {
+        /// <summary>
+        /// Encodes the given text to Base64, using the given <see cref="Encoding"/> (default: <see cref="Encoding.UTF8"/>
+        /// </summary>
+        /// <param name="textToEncode"></param>
+        /// <param name="encoding">Default is <see cref="Encoding.UTF8"/>.</param>
+        /// <param name="insertLineBreaks">Inserts line breaks after every 76 characters in the string representation.</param>
+        /// <returns>The encoded string.</returns>
+        public static string? ToBase64(this string? textToEncode, Encoding? encoding = null, bool insertLineBreaks = true)
```

Comment:
Hint, you could add an attribute that says: only returns null if input is null

Reply:
Yes, `NotNullIfNotNull("textToEncode")`, which is only available in NetStandard2.1